cognite/client/_api/entity_matching.py

```diff
@@ -0,0 +1,220 @@
+        true_matches: List[Union[Dict, Tuple[Union[int, str], Union[int, str]]]] = None,
+        match_fields: List[Tuple[str, str]] = None,
+        feature_type: str = None,
+        classifier: str = None,
+        ignore_missing_fields: bool = False,
+        name: str = None,
+        description: str = None,
+        external_id: str = None,
+    ) -> EntityMatchingModel:
+        """Fit entity matching model.
+        Note: All users on this CDF subscription with assets read-all capability in the project, are able to access the data sent to this endpoint.
+
+        Args:
+            sources: entities to match from, should have an 'id' field. Tolerant to passing more than is needed or used (e.g. json dump of time series list)
+            targets: entities to match to, should have an 'id' field.  Tolerant to passing more than is needed or used.
+            true_matches: Known valid matches given as a list of dicts with keys 'sourceId', 'sourceExternalId', 'sourceId', 'sourceExternalId'). If omitted, uses an unsupervised model.
+             A tuple can be used instead of the dictionary for convenience, interpreted as id/externalId based on type.
+            match_fields: List of (from,to) keys to use in matching. Default in the API is [('name','name')]
+            feature_type (str): feature type that defines the combination of features used, see API docs for details.
+            classifier (str): classifier used in training. Currently undocumented in API.
+            ignore_missing_fields (bool): whether missing data in keyFrom or keyTo should return error or be filled in with an empty string. Currently undocumented in API.
+            name (str): Optional user-defined name of model.
+            description (str): Optional user-defined description of model.
+            external_id (str): Optional external id. Must be unique within the project.
+        Returns:
+            EntityMatchingModel: Resulting queued model."""
+
+        if match_fields:
+            match_fields = [ft if isinstance(ft, dict) else {"source": ft[0], "target": ft[1]} for ft in match_fields]
+        if true_matches:
+            true_matches = [convert_true_match(true_match) for true_match in true_matches]
+        response = self._post(
+            self._RESOURCE_PATH + "/",
+            json={
+                "name": name,
+                "description": description,
+                "externalId": external_id,
+                "sources": EntityMatchingModel.dump_entities(sources),
+                "targets": EntityMatchingModel.dump_entities(targets),
+                "trueMatches": true_matches,
+                "matchFields": match_fields,
+                "featureType": feature_type,
+                "classifier": classifier,
+                "ignoreMissingFields": ignore_missing_fields,
+            },
+        )
+        return self._LIST_CLASS._RESOURCE._load(response.json(), cognite_client=self._cognite_client)
```

Comment:
Not sure how I feel about this - do you think it would look better or worse if we unified a lot of the business logic massaging into the constructor of the EntityMatchingModel class? That way, we could allow the users to create these things with the same flexibility, but we could use more standard patterns for implementing functionality such as this.

Reply:
not sure what you mean by 'a lot of business logic', but we're a bit unusual in that we don't send the sources/targets back.